src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -284,11 +276,16 @@ private static void TryFormatCSharpBlockStructure(FormattingContext context, Lis
             !directive.ContainsDiagnostics &&
             directive.DirectiveDescriptor?.Kind == DirectiveKind.CodeBlock)
         {
+            // If we're formatting a @code or @functions directive, the user might have indicated they always want a newline
+            var forceNewLine = _optionsMonitor.CurrentValue.CodeBlockBraceOnNextLine &&
+                directive.Body is RazorDirectiveBodySyntax { Keyword: { } keyword } &&
+                keyword.GetContent() is "code" or "functions";
```

Comment:
❓ are there not constants for these keywords?

Reply:
Good call. There is a directive class we can use. Have to lose the pattern, but thats okay.